src/renderer/components/+catalog/catalog-add-button.tsx

```diff
@@ -0,0 +1,77 @@
+import "./catalog-add-button.scss";
+import React from "react";
+import { SpeedDial, SpeedDialAction } from "@material-ui/lab";
+import { Icon } from "../icon";
+import { disposeOnUnmount, observer } from "mobx-react";
+import { observable, reaction } from "mobx";
+import { autobind } from "../../../common/utils";
+import { CatalogCategory, CatalogEntityAddMenuContext, CatalogEntityContextMenu } from "../../api/catalog-entity";
+import { EventEmitter } from "events";
+import { navigate } from "../../navigation";
+
+export type CatalogAddButtonProps = {
+  category: CatalogCategory
+};
+
+@observer
+export class CatalogAddButton extends React.Component<CatalogAddButtonProps> {
+  @observable protected isOpen = false;
+  protected menuItems = observable.array<CatalogEntityContextMenu>([]);
+
+  componentDidMount() {
+    const { category } = this.props;
+
+    disposeOnUnmount(this, [
+      reaction(() => category, (category) => {
```

Comment:
For some reason this does not react at all when category changes.

Reply:
Just use `this.props.category` directly here without upper line
```
const { category } = this.props;
```